MSAL/src/MSALErrorConverter.m

```diff
@@ -44,29 +44,35 @@ + (void)initialize
-                                   @(MSIDErrorServerInvalidResponse) : @(MSALErrorInvalidResponse),
-                                   @(MSIDErrorDeveloperAuthorityValidation) : @(MSALErrorFailedAuthorityValidation),
-                                   @(MSIDErrorServerRefreshTokenRejected) : @(MSALErrorAuthorizationFailed),
-                                   @(MSIDErrorServerOauth) : @(MSALErrorAuthorizationFailed),
-                                   @(MSIDErrorUnsupportedFunctionality): @(MSALErrorInternal)
+                                   // Authority Validation
+                                   @(MSIDErrorAuthorityValidation) : @(MSALErrorFailedAuthorityValidation),
+                                   // Interactive flow
+                                   @(MSIDErrorAuthorizationFailed) : @(MSALErrorAuthorizationFailed),
+                                   @(MSIDErrorUserCancel) : @(MSALErrorUserCanceled),
+                                   @(MSIDErrorSessionCanceledProgramatically) : @(MSALErrorSessionCanceled),
+                                   @(MSIDErrorInteractiveSessionStartFailure) : @(MSALErrorInternal),
+                                   @(MSIDErrorInteractiveSessionAlreadyRunning) : @(MSALErrorInteractiveSessionAlreadyRunning),
+                                   @(MSIDErrorNoMainViewController) : @(MSALErrorNoViewController),
                                    },
                            MSIDOAuthErrorDomain:@{
-                                   @(MSIDErrorInvalidRequest) :@(MSALErrorInvalidRequest),
-                                   @(MSIDErrorInvalidClient) : @(MSALErrorInvalidClient),
-                                   @(MSIDErrorInvalidGrant) : @(MSALErrorInvalidParameter),
-                                   @(MSIDErrorInvalidParameter) : @(MSALErrorInvalidParameter),
-                                   @(MSIDErrorServerRefreshTokenRejected) : @(MSALErrorAuthorizationFailed),
+                                   @(MSIDErrorInteractionRequired) : @(MSALErrorInteractionRequired),
                                    @(MSIDErrorServerOauth) : @(MSALErrorAuthorizationFailed),
+                                   @(MSIDErrorServerInvalidResponse) : @(MSALErrorInvalidResponse),
+                                   @(MSIDErrorServerRefreshTokenRejected) : @(MSALErrorAuthorizationFailed),
+                                   @(MSIDErrorServerInvalidRequest) :@(MSALErrorInvalidRequest),
+                                   @(MSIDErrorServerInvalidClient) : @(MSALErrorInvalidClient),
+                                   @(MSIDErrorServerInvalidGrant) : @(MSALErrorInvalidGrant),
+                                   @(MSIDErrorServerInvalidScope) : @(MSALErrorInvalidScope),
+                                   @(MSIDErrorServerInvalidState) : @(MSALErrorInvalidState),
+                                   @(MSIDErrorServerNonHttpsRedirect) : @(MSALErrorInternal)
```

Comment:
Don't we have a corresponding MSAL error? If not, let's add it, as it sounds important to call out this specific reason.

Reply:
added corresponding error for msal. also filed an issue to improve refactoring #285